pkg/apis/apps/validation/validation.go

```diff
@@ -69,6 +70,33 @@ func ValidatePodTemplateSpecForStatefulSet(template *api.PodTemplateSpec, select
 	return allErrs
 }
 
+func ValidatePersistentVolumeClaimDeletePolicy(policy apps.PersistentVolumeClaimDeletePolicy, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	switch {
+	case policy == apps.RetainPersistentVolumeClaimDeletePolicyType:
+	case policy == apps.DeletePersistentVolumeClaimDeletePolicyType:
+	case policy == "": // Same as the default retain value.
+	default:
+		allErrs = append(allErrs, field.NotSupported(fldPath, policy, []string{string(apps.RetainPersistentVolumeClaimDeletePolicyType), string(apps.DeletePersistentVolumeClaimDeletePolicyType)}))
+	}
+	return allErrs
+}
+
+func ValidatePersistentVolumeClaimPolicy(policy *apps.StatefulSetPersistentVolumeClaimPolicy, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if utilfeature.DefaultFeatureGate.Enabled(features.StatefulSetAutoDeletePVC) && policy == nil {
+		allErrs = append(allErrs, field.Required(fldPath, ""))
```

Comment:
set a message.

But maybe it's fine to remove this check, as defaulting already looks at the feature gate to not leave it empty.

Reply:
We still need to verify the fields inside of policy, and it seems dangerous to leave open the possibility of a nil pointer exception.